SimTracker/SiPixelDigitizer/plugins/SiPixelDigitizerAlgorithm.cc

```diff
@@ -1366,7 +1401,7 @@ void SiPixelDigitizerAlgorithm::make_digis(float thePixelThresholdInE,
 
     // Do only for pixels above threshold
 
-    if( signalInElectrons >= thePixelThresholdInE) { // check threshold
+    if( signalInElectrons >= thePixelThresholdInE && signalInElectrons > 0.) { // check threshold, always reject killed (0-charge) digis
```

Comment:
The added check is needed for the case that `NoiseInElectrons <= 0.` in the configuration.

Reply:
@makortel , is it possible that thePixelThresholdInE < 0.0? if not, I cannot understand the change.

Reply:
It can be exactly `0.0` if `NoiseInElectrons` configuration parameter is `<= 0`. But `signalInElectrons >= 0.0` is not enough to reject killed digis as for them `signalInElectrons` is `0.0`. The reason is the same for which #21175 was needed for premixing.

In practice, for normal configuration the added check has no effect. It is needed if one wants to check that it is possible to get exactly same results with this PR as in the baseline (see PR description).

I don't feel strongly about the check though, given that it is only for special tests that are probably relevant only during the review of this PR. On the other hand, I think seeing the `signalInElectrons > 0.` requirement explicitly instead of relying `thePixelThesholdInE > 0.` makes it easier to understand that `signalInElectrons == 0.0` cases are intended to be removed.